airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/jobs/registry.py

```diff
@@ -24,6 +24,35 @@
 )
 
 
+@op(required_resource_keys={"all_metadata_file_blobs"})
+def remove_stale_metadata_partitions_op(context):
+    """
+    This op is responsible for polling for new metadata files and adding their etag to the dynamic partition.
+    """
+    all_metadata_file_blobs = context.resources.all_metadata_file_blobs
+    partition_name = registry_entry.metadata_partitions_def.name
+
+    all_fresh_etags = [blob.etag for blob in all_metadata_file_blobs]
+
+    all_etag_partitions = context.instance.get_dynamic_partitions(partition_name)
+
+    stale_etags = [etag for etag in all_etag_partitions if etag not in all_fresh_etags]
+    context.log.info(f"Stale etags found: {stale_etags}")
+    for stale_etag in stale_etags:
```

Comment:
nit: you can avoid one iteration if you filter and delete in a single iteration on `all_etag_partitions`

Reply:
Ah great call. Will change